sds/src/match_validation/config.rs

```diff
@@ -74,9 +74,18 @@ pub struct HttpValidatorConfig {
 }
 
 impl HttpValidatorConfig {
-    pub fn new(endpoint: &str) -> Self {
+    pub fn new(endpoint: &str, hosts: Vec<String>) -> Self {
+        // Replace $HOSTS in endpoint and build the endpoints vector
+        let mut endpoints = vec![];
+        for host in hosts {
+            endpoints.push(endpoint.replace("$HOSTS", &host));
```

Comment:
From a pure API point of view, I would expect to write `$HOST` and not `$HOSTS` because in the endpoint, I expect a single host to replace `$HOST`